### ISSUE TITLE
speedup tests

### ISSUE THREAD

**Issue by lbergelson** [broadinstitute/hellbender#221]
currently the travis build is at ~10minutes

There are several options, and we should probably do all of them:

1. Speedup individual tests:
  The top offenders are:
   `SplitNCigarReadsIntegrationTest`: this 1 test takes a minute, we can probably do something about this
   `MarkDuplicatesIntegrationTest`: 56 tests each taking ~1 second, unclear what we could do
   `AnalyzeCovariatesIntegrationTests` 13 tests taking ~40 seconds. Some take longer than others, not sure what we can do about these
`CachingIndexedFastaSequenceFileUnitTest	` 21 tests taking a minute.  Lets push this to htsjdk.

2.  We lose several minutes installing R libraries (2-3).  We could parallelize our travis build and split it into 2 builds, so that 1 branch of the build installs R libraries and then runs the tests that depend on those, and the other branch runs all the other tests.  This would probably make the R installation effectively free provided we have sufficient worker nodes.

3.  Run tests in parallel.  Travis gives us more than 1 core.  I tested running with cores set to 2, and the actual test time dropped nearly in half.

**Comment by akiezun** [comment]
10 minutes does not bother me.
